c_src/kos_msg_nif.c

```diff
@@ -155,10 +157,12 @@ thr_main(void* obj)
     status = setup_msg_server_transport(p_env);
     kos_assert_created(status, NULL);
 
-    server_reply_cap = kos_cnode_cap(p_env->p_cnode, KOS_THREAD_SLOT_REPLY);
-
     while(true){
-      seL4_MessageInfo_t sel4_msg = seL4_Recv(_server.transport.ep_cptr, NULL, _server.reply_cptr);
+#ifdef CONFIG_KERNEL_MCS
```

Comment:
To avoid this if def you can use `kos_ep_msg_info_t kos_cap_recv(kos_cptr_t c, kos_cap_badge_t* b, kos_cptr_t r)` which will not use the reply cap in non-MCS builds. This will give you a different return type but `kos_ep_msg_info_get_label` and `kos_ep_msg_info_get_length` can be used below